### PR TITLE
Apply Zend Coding Standard 2

### PR DESCRIPTION
**Pull request by xtreamwayz**
```
PHPCBF RESULT SUMMARY
----------------------------------------------------------------------
A TOTAL OF 300 ERRORS WERE FIXED IN 59 FILES
----------------------------------------------------------------------
```

Errors and warnings after phpcbf
```
FILE: /c/Users/xtreamwayz/projects/xtreamwayz/zend-diactoros/src/Exception/InvalidStreamPointerPositionException.php
--------------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------
 17 | WARNING | Possible useless method overriding detected
    |         | (Generic.CodeAnalysis.UselessOverridingMethod.Found)
--------------------------------------------------------------------------------------------------------------------


FILE: /c/Users/xtreamwayz/projects/xtreamwayz/zend-diactoros/src/Exception/UploadedFileAlreadyMovedException.php
-----------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------
 17 | WARNING | Possible useless method overriding detected
    |         | (Generic.CodeAnalysis.UselessOverridingMethod.Found)
-----------------------------------------------------------------------------------------------------------------


FILE: /c/Users/xtreamwayz/projects/xtreamwayz/zend-diactoros/src/Uri.php
----------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------------
 649 | ERROR | Class Uri contains unused private method urlEncodeChar().
     |       | (SlevomatCodingStandard.Classes.UnusedPrivateElements.UnusedMethod)
----------------------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

File: test/ServerRequestFactoryTest.php [hunk 1]
```diff
@@ -382,22 +370,22 @@ public function testCanCreateServerRequestViaFromGlobalsMethod()
...
+        $query['query'] = true;
+        $body['body'] = true;
+        $files = [
+            'files' => [
+                'tmp_name' => 'php://temp',
+                'size'     => 0,
+                'error'    => 0,
+                'name'     => 'foo.bar',
+                'type'     => 'text/plain',
+            ],,
```
Comment (xtreamwayz) [inline test/ServerRequestFactoryTest.php]: Something goes wrong here. Need to check this out.

File: src/Exception/InvalidStreamPointerPositionException.php [hunk 2]
```diff
@@ -17,7 +17,7 @@ class InvalidStreamPointerPositionException extends RuntimeException implements
     public function __construct(
         string $message = 'Invalid pointer position',
         $code = 0,
-        Throwable $previous = null
+        ?Throwable $previous = null
```
Comment (webimpress) [inline src/Exception/InvalidStreamPointerPositionException.php]: What's the point adding `?` if the default value is `null`?

It was fixed automatically or manually? If automatically I can see potential issues with code which implement interface/extend other class on which we do not have control.

Comment (xtreamwayz) [reply src/Exception/InvalidStreamPointerPositionException.php]: I did not make any changes manually. This is all phpcbf.

**Comment by xtreamwayz** [comment]
Issue found and reported: https://github.com/slevomat/coding-standard/issues/477

Turns out a rule is importing functions which are in the same namespace.

File: src/Exception/InvalidStreamPointerPositionException.php (hunk 2 above)
Comment (weierophinney) [reply src/Exception/InvalidStreamPointerPositionException.php]: @webimpress This makes sense to me. I was surprised the first time I saw that setting a default value of `null` on a typehinted argument meant that it was considered nullable; adding the explicit notation in the type declaration makes that abundantly clear to users. As such, I definitely support this sort of change.

**Pull request closed without merging**